### PR TITLE
Adds the medical cyborg surgical processor upgrade

### PR DESCRIPTION
**Pull request by FrostyFridge**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Frosty Fridge
add: Added the Surgical Processor upgrade for medical cyborgs. Scan surgery disks or an operating computer to be able to initiate advanced procedures.
tweak: Cyborgs can now perform surgery steps that do not require an instrument.
/:cl:

Adds the Surgical Processor upgrade for medical cyborgs, available from exosuit fabricators after 'Cyborg Utilities: Medical' is researched. This upgrade allows medical cyborgs to scan surgery disks, or copy procedures from an operating computer. The cyborg can then initiate scanned procedures. Cyborgs can also now perform surgery steps that have no instrument requirement.

Medical cyborgs were generally unable to perform any advanced surgeries outside of some specific circumstances. This update allows cyborgs to make use of the advanced surgeries. Some advanced surgeries have steps that require a hand without any tools; this update allows cyborgs to perform these steps as well. Note that surgeries must be obtained through research or a disk somehow before the cyborgs can scan them.

First time using DM, so let me know if any of the code is too snowflakey, something is broken, or if there's a better way to do these things. Thanks!

### CODE REVIEW

**Comment by GunHog** [comment]
:warning: 
https://github.com/tgstation/tgstation/pull/39451 has removed surgical disks!
:warning:

You will need to modify your upgrade item to sync with the research network, or allow it to copy surgeries from an operating computer.

**Comment by SpaceManiac** [comment]
You can keep the disk handling - there's still the traitor brainwashing disk - but you'll want additional handling. Best bet is probably clicking with the module on a surgery computer to load its surgeries.

**Comment by Ratvar** [comment]
Or automatically unlocking when R&D researches them?

File: code/modules/surgery/tools.dm [hunk 1]
```diff
@@ -235,3 +235,28 @@
...
 	return
+
+/obj/item/surgical_processor //allows medical cyborgs to scan and initiate advanced surgeries
+	name = "Surgical Processor"
+	desc = "A device for scanning and initiating surgeries from a disk or operating computer."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "spectrometer"
+	var/list/advanced_surgeries = list()
+	
+/obj/item/surgical_processor/afterattack(obj/item/O, mob/user, proximity)
```
Comment (SpaceManiac) [inline code/modules/surgery/tools.dm]: Scanning the surgical computer with the module still slaps it (attack animation, sound).

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/modules/surgery/tools.dm (hunk 1 above)
Comment (FrostyFridge) [reply code/modules/surgery/tools.dm]: Would changing the proc from afterattack() to attack() be an acceptable way to fix this?

Comment (FrostyFridge) [reply code/modules/surgery/tools.dm]: nevermind, forgot about itemflags

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**